### ISSUE TITLE
Tutorial matrix

### ISSUE THREAD

**Issue by joostveenema** [FluxML/model-zoo#280]
This is issue is mainly to get a discussion started about what the (contentwise) end state of the model zoo should be after the refresh. Not to skip steps in the list from https://github.com/FluxML/ML-Coordination-Tracker/issues/9, but we better have this at least a bit settled before starting to convert, add or trim any of the tutorials.

Below I started a matrix to see what the coverage of problems, datasets and highlighted features will be. Ideally we fill it mostly with the already existing models in the zoo and create some new ideas for tutorials to fill any gaps. The rows I already added are merely suggestions, feel free to edit.

problem | model | dataset | highlighted features (max 2)
-----------|----------|-----------|-----------------------------------
Linear regression | - | MLDatasets.BostonHousing | Low level API
Image classification | LeNet5 | MLDatasets.MNIST | Model saving/loading
Language detection | LSTM | Custom | Custom dataset
Transfer learning | ResNet | ? | Transfer learning

**Comment by DhairyaLGandhi** [comment]
Before we get into this, could we get an idea of what models remain to be updated.

**Comment by joostveenema** [comment]
The list of open issues is in https://github.com/FluxML/model-zoo/issues/266, in short: they all run or need major work to get them running. 

But I strongly prefer to have the direction settled before solving any of these remaining issues. The problems are often in parts that are not really necessary and could thus be resolved by removing things from the model. 

Take for example the language detection model. The dataset handling should be improved, but there are two options: use a different, more "standard" dataset or stay with the custom dataset creation. If in the end we don't want to focus on the creation of a dataset, this part in the code should just be removed and it would be a waste of effort trying to improve it now.

**Comment by DhairyaLGandhi** [comment]
Sure. I would add the following:

Vae - custom loss
Cppn - training params directly
Conv - custom training loop
Cifar10 - standard datasets, pretrained models
Vgg - sequential chain

We should add examples from the raytracer and flux3d for some more recent dP problems, even as stubs to point to the right places cc @avik-pal @nirmal-suthar

**Comment by avik-pal** [comment]
For reference:
1. Examples for RayTracer.jl --> https://github.com/avik-pal/RayTracer.jl#tutorials
2. Examples for Flux3D.jl --> https://github.com/FluxML/Flux3D.jl#examples

I think stubs to point to the right examples is the right way, else it becomes very tedious to update the same example at multiple places (after any breaking change).

**Comment by pevnak** [comment]
@darsnack asked me to put https://github.com/pevnak/Mill.jl and https://github.com/pevnak/JsonGrinder.jl here.

Using the matrix from the above:
**problem:**  hierarchical multiple-instance learning problems, under which you can imagine JSONs, ProtoBuffers, etc. To our know
**model:** all classes of models as defined in Tomáš Pevný , Vojtěch Kovařík (2019), Approximation capability of neural networks on spaces of probability measures and tree-structured domains, arXiv:1906.00764 
**Dataset** Recipes, which are very simple, or https://github.com/endgameinc/ember, which is gigantic
**Features** a step towards AutoML on general data
